request.go

```diff
@@ -203,6 +204,14 @@ func WithBytesPayload(payload []byte) RequestOption {
 	}
 }
 
+// WithRetryOnEOFError adds the io.EOF error to the retry loop
```

Comment:
I think that this comment should be more explicit, about how this is what shows up for bad connections. I even think some links to the actual GitHub issues for Go on the stdlib would be useful.

I also wonder about dropping the `With` here, and just calling it `RetryOnEOFError`

Reply:
I'll add the more explicit comment. I think we'll table the rename of the RequestOption for now. Currently all Options in the fetcher code begin with `With`.